pkg/loop/internal/ccip/test/exec_provider.go

```diff
@@ -60,106 +65,169 @@ type staticExecProvider struct {
-func (s *staticExecProvider) NewOffRampReader(ctx context.Context, addr ccip.Address) (ccip.OffRampReader, error) {
-	return s.offrampreader, nil
+func (s staticExecProvider) NewOffRampReader(ctx context.Context, addr ccip.Address) (ccip.OffRampReader, error) {
+	return s.offRampReader, nil
 }
 
 // NewOnRampReader implements ExecProviderEvaluator.
-func (s *staticExecProvider) NewOnRampReader(ctx context.Context, addr ccip.Address) (ccip.OnRampReader, error) {
-	return s.onrampreader, nil
+func (s staticExecProvider) NewOnRampReader(ctx context.Context, addr ccip.Address) (ccip.OnRampReader, error) {
+	return s.onRampReader, nil
 }
 
 // NewPriceRegistryReader implements ExecProviderEvaluator.
-func (s *staticExecProvider) NewPriceRegistryReader(ctx context.Context, addr ccip.Address) (ccip.PriceRegistryReader, error) {
-	panic("unimplemented")
+func (s staticExecProvider) NewPriceRegistryReader(ctx context.Context, addr ccip.Address) (ccip.PriceRegistryReader, error) {
+	return s.priceRegistryReader, nil
 }
 
 // NewTokenDataReader implements ExecProviderEvaluator.
-func (s *staticExecProvider) NewTokenDataReader(ctx context.Context, tokenAddress ccip.Address) (ccip.TokenDataReader, error) {
+func (s staticExecProvider) NewTokenDataReader(ctx context.Context, tokenAddress ccip.Address) (ccip.TokenDataReader, error) {
 	panic("unimplemented")
 }
 
 // NewTokenPoolBatchedReader implements ExecProviderEvaluator.
-func (s *staticExecProvider) NewTokenPoolBatchedReader(ctx context.Context) (ccip.TokenPoolBatchedReader, error) {
+func (s staticExecProvider) NewTokenPoolBatchedReader(ctx context.Context) (ccip.TokenPoolBatchedReader, error) {
 	panic("unimplemented")
 }
 
 // OffchainConfigDigester implements ExecProviderEvaluator.
-func (s *staticExecProvider) OffchainConfigDigester() libocr.OffchainConfigDigester {
+func (s staticExecProvider) OffchainConfigDigester() libocr.OffchainConfigDigester {
 	return s.offchainDigester
 }
 
 // Ready implements ExecProviderEvaluator.
-func (s *staticExecProvider) Ready() error {
+func (s staticExecProvider) Ready() error {
 	return nil
 }
 
 // SourceNativeToken implements ExecProviderEvaluator.
-func (s *staticExecProvider) SourceNativeToken(ctx context.Context) (ccip.Address, error) {
+func (s staticExecProvider) SourceNativeToken(ctx context.Context) (ccip.Address, error) {
 	panic("unimplemented")
 }
 
 // Start implements ExecProviderEvaluator.
-func (s *staticExecProvider) Start(context.Context) error {
+func (s staticExecProvider) Start(context.Context) error {
 	return nil
 }
 
 // AssertEqual implements ExecProviderTester.
-func (s *staticExecProvider) AssertEqual(ctx context.Context, t *testing.T, other types.CCIPExecProvider) {
-	// TODO BCF-2979 other components of exec provider
-	assert.NoError(t, s.Evaluate(ctx, other))
-}
+func (s staticExecProvider) AssertEqual(ctx context.Context, t *testing.T, other types.CCIPExecProvider) {
+	t.Parallel()
+	t.Run("StaticExecProvider", func(t *testing.T) {
```

Comment:
Is the subtest meant to be parallel?
```suggestion
	t.Run("StaticExecProvider", func(t *testing.T) {
		t.Parallel()
```

Reply:
no, just a mistake. fixed, thanks

Reply:
This still looks the same to me?

Reply:
now it fixed. parallel wasn't supposed to be there. thought i had deleted it